contents/java/JUnit5.md

```diff
@@ -0,0 +1,259 @@
+<frontmatter>
+  title: Introduction to JUnit 5
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Testing in Java with JUnit 5
+
+Author: [Lin Si Jie](https://github.com/sijie123)
+
+Reviewers: [Aadyaa Maddi](https://github.com/amad-person), [Marvin Chin](https://github.com/marvinchin)
+
+## What is JUnit 5?
+JUnit is an open-source automated testing framework for Java. Using its vast API library, you can easily write and run tests to ensure correctness in your software engineering project. JUnit 5 is the latest iteration of the JUnit project, supporting multiple new features in Java (such as Lambdas) and adopting a new <popover effect="fade" content="think: OOP for testing" placement="top">Extension-based model</popover> to make writing tests easier, cleaner, and more modular.
+
+## Why do automated testing?
+There are many reasons why automated testing is important for any software engineering project.
+
+<trigger for="modal:automatedTesting2103" trigger="click">Automated testing</trigger>
+* is the first line of defence against bugs
+* protects against <popover effect="fade" content="changes that unintentionally break existing functionality" placement="top">regression</popover>
+* checks against expected output, preventing errors arising from <popover effect="fade" content="looking manually at output" placement="top">LMAO</popover> or <popover effect="fade" content="looking manually for abnormal output" placement="top">LMFAO</popover>. ([Acronym source](https://stackoverflow.com/questions/10858990/why-use-junit-for-testing))
+* ultimately, gives the developer confidence that their code works correctly.
+
+In fact, testing is so powerful and important that an entire <trigger for="modal:agileDev" trigger="click">agile development process</trigger> named <trigger for="modal:tdd" trigger="click">Test Driven Development</trigger> is designed with tests as the centerpiece of software development.
```

Comment:
I think we are straying away from the topic. Automated testing can be an article of its own. In here, just focus on JUnit.